File: 1_Java_Examples/2_Shapes_LSP/Rectangle.java
public class Rectangle {
	
	private int x, y, horizontalLength, verticalLength;
	
	public Rectangle(int x, int y, int horizontalLength, int verticalLength) {
		this.x = x;
		this.y = y;
		this.horizontalLength = horizontalLength;
		this.verticalLength = verticalLength;
	}
	
	public int getHorizontalLength() {
		return this.horizontalLength;
	}
	
	public int getVerticalLength() {
		return this.verticalLength;
	}
	
	public int computeArea() {
		return horizontalLength * verticalLength;
	}
	
	public int computePerimenter() {
		return horizontalLength * 2 + verticalLength * 2;
	}
	
	public void setSideLengths(int horizontalLength, int verticalLength) {
		this.horizontalLength = horizontalLength;
		this.verticalLength = verticalLength;
	}
	
}

File: 2_Ruby_Examples/2_Shapes_LSP/rectangle.rb
class Rectangle
	
	attr_accessor :horizontal_length, :vertical_length
	
	def initialize x, y, horizontal_length, vertical_length
		@x = x;
		@y = y;
		@horizontal_length = horizontal_length;
		@vertical_length = vertical_length;
	end
	
	def compute_area
		@horizontal_length * @vertical_length;
	end
	
	def compute_perimenter
		@horizontal_length * 2 + @vertical_length * 2;
	end
	
	def set_side_lengths horizontal_length, vertical_length
		@horizontal_length = horizontal_length;
		@vertical_length = vertical_length;
	end
	
end

